### DISCUSSION TITLE
Is it possible to disable autolinking in remark-gfm?

### DISCUSSION THREAD

**Discussion by Trott** [remarkjs/remark#862] [Q&A] [answered]
Is it possible to use remark-gfm but disable autolinking?

**What I'm seeing**

I want to use GFM, but mostly for tables and maybe strikethrough. I don't mind checkboxes one way or the other. Autolinking would be fine, but I find that it links a lot of things that I don't intend to be links (and that GitHub doesn't link either--so maybe this is a bug?).

So, if I have a GFM table with the columns not aligned, remark-gfm aligns them if I parse and then stringify. Great.

But if I have something like this:

```markdown
* remark-gfm@2.0.0
```

Here's how GitHub renders it:

* remark-gfm@2.0.0

(Specifically, it does not turn the text into a link.)

But if I parse and stringify that text with remark-gfm....

```console
echo '* remark-gfm@2.0.0' | remark -u remark-gfm 
*   <remark-gfm@2.0.0>
<stdin>: no issues found
```

It surrounds it in `<` and `>`, making it a link.

It's possible I'm misunderstanding parse and stringify. I'm hoping to use `remark-gfm` to basically format my markdown consistently by parsing it and stringifying it. But this behavior kind of makes it a non-starter for changelog files with hundreds of strings in that format.

So options I've considered:

* Find a way to use `remark-gfm` but disable autolinking, or at least autolinking of email addresses. (That's what I'm asking about in this question.)
* Open an issue and/or dive in and see if I can fix the issue myself. Trying to reverse engineer GitHub's rendering into a parser algorithm or regular expression is not my idea of a good time, but maybe someone else would be psyched to do it.
* Open an issue and/or implement a feature to disable autolinking via an option in `remark-gfm` if that option doesn't already exist.
* Copy over remark-gfm and remove the autolinking myself. (I've done this as a proof of concept, but I would prefer not to maintain my own fork of remark-gfm.)

Thanks for taking the time to read and for providing this robust ecosystem for people like me to use for free. ❤️

**Comment by Trott** [comment]
I'm not sure if it's the right/complete solution, but https://github.com/syntax-tree/mdast-util-gfm-autolink-literal/pull/5 seems to fix this issue and doesn't break any existing tests, so hopefully it's on the right track at least?

**Comment by wooorm** [comment, marked as answer]
> Is it possible to use remark-gfm but disable autolinking?

You could do what `remark-gfm` does — add micromark and mdast-util-{to,from} extensions/utilities — but without `micromark-extension-gfm-autolink-literal` and `mdast-util-gfm-autolink-literal`.

> but I find that it links a lot of things that I don't intend to be links (and that GitHub doesn't link either--so maybe this is a bug?)

These are indeed bugs! I’ve spent a *lot* of time trying to reverse engineer what GitHub does. But it’s hard because it’s not specified *and because they use different algorithms in different places*.
I’d appreciate hearing about all the cases where our algorithm doesn’t match theirs.

> I'm hoping to use remark-gfm to basically format my markdown consistently by parsing it and stringifying it. But this behavior kind of makes it a non-starter for changelog files with hundreds of strings in that format.

Yep, that’s what it’s supposed to do. And indeed a bug.

> Open an issue and/or dive in and see if I can fix the issue myself. Trying to reverse engineer GitHub's rendering into a parser algorithm or regular expression is not my idea of a good time, but maybe someone else would be psyched to do it.

I want to do this!

**Reply by wooorm** [reply]
Solved the bug in https://github.com/syntax-tree/mdast-util-gfm-autolink-literal/commit/7555d45f9ad4ecdd578ffedf3cbc4b01d3ae0135. Thanks @Trott for the detailed info, the hint for a fix, and the kind words! 🤗
